Adds new security item: Anti-Magic Collars

cowbot92 opened the pull request:
Special thanks to @Chubbygummibear for keeping me sane through all of this

# Document the changes in your pull request

Adds a new security item to counter caught magic users (heretics & wizards only ATM). This Item is called the Anti-Magic Collar, and it is a unremovable (through natural means) collar that is meant to dampen the danger that both Heretics & Wizards may cause to the station after being caught.  When equipped, the collar no longer allows you to cast magic spells that are blocked by anti-magic (without giving you full anti-magic protection) 

For now there are 3 of them in the security vendor for 100 credits each (or free if you're sec) but this can be moved/adjusted as needed.

Additionally fixes Heretic spells so that you cannot cast them if you have anti-magic, which is the whole point of this, Most wizard spells will already fall under being affected by this as well, as a bonus.

# Why is this good for the game?

I believe this is good for the game for a number of reasons: Firstly, it sucks when you get caught early, round removed, and then get to watch as a ghost all round instead of getting to participate in SOME way besides "oh hey here's a blob spawn." Now if you get caught, as long as your crimes don't outweigh anything other than "this guy used magic once" security might think twice before just shoving you in the morgue. Bonus points for getting on the AI's good side if it's lawset isn't supposed to let them keep you dead.

In the end it's an alternative I would like the server to have to just straight up round ending people who don't deserve it.

# Testing

All testing for heretic spells + being able to purchase the collars was done in this video

https://github.com/yogstation13/Yogstation/assets/75333826/18672258-fb80-44f8-8423-b615d8a05193

# Spriting

New sprites, by me! **Updated

![image](https://github.com/yogstation13/Yogstation/assets/75333826/353c6c4a-98ce-4978-beb7-6ed240a95619)

![image](https://github.com/yogstation13/Yogstation/assets/75333826/40ff16ac-6af8-45ac-9ed5-c66bd5360268)

![anti_magic_collar](https://github.com/yogstation13/Yogstation/assets/75333826/2a74174c-59ea-4e50-b257-7d9a6618c812)

![anti_magic_collar_on_person](https://github.com/yogstation13/Yogstation/assets/75333826/4d7b9335-714e-4f2f-80d9-ce61fbb4e7ad)

# Wiki Documentation

I will be adding this to the wiki as well.

# Changelog

:cl:  
rscadd: Adds new anti-magic item for security
tweak: Tweaks around some heretic spell reqs
imageadd: Adds new sprites for anti-magic collar
/:cl:

cowbot92 committed:
cool

yeah

cowbot92 committed:
yep thats all of it

yep

cowbot92 committed:
Merge remote-tracking branch 'upstream/master' into AntiMagicCollars

cowbot92 committed:
why must I forget to change this

im awful

cowbot92 committed:
Merge remote-tracking branch 'upstream/master' into AntiMagicCollars

cowbot92 committed:
Merge remote-tracking branch 'upstream/master' into AntiMagicCollars

cowbot92 committed:
updated sprites upon request

its blue now

cuackles reviewed (approved):
L to helios

JohnFulpWillard reviewed (changes requested):
This has to be reworked to use the newer anti magic system (that splits anti-magic checks between using and being used on)

cowbot92 committed:
Merge remote-tracking branch 'upstream/master' into AntiMagicCollars

cowbot92 committed:
update removing all my dumb shit

thanks though chubby

cowbot92 committed:
ONE space

the one space is real

JohnFulpWillard reviewed (changes requested):
All the new ``SPELL_REQUIRES_NO_ANTIMAGIC`` is intended, right?

cowbot92 committed:
pupdate

wahoo

cowbot92 committed:
Adds the thing

does it too

cowbot92 committed:
Adds notices for locking/unlocking

Now the collars will tell the user when it is locked around their neck, and tell them when it is unlocked via the console.

cowbot92 committed:
this iddiot forgot to remove low memory mode

god damnit
